ddl/ddl_worker.go

```diff
@@ -427,15 +428,37 @@ func toTError(err error) *terror.Error {
 
 // waitSchemaChanged waits for the completion of updating all servers' schema. In order to make sure that happens,
 // we wait 2 * lease time.
-func (d *ddl) waitSchemaChanged(waitTime time.Duration) {
+func (d *ddl) waitSchemaChanged(waitTime time.Duration, latestSchemaVersion int64) {
 	if waitTime == 0 {
 		return
 	}
 
-	select {
-	case <-time.After(waitTime):
-	case <-d.quitCh:
+	if !ChangeOwnerInNewWay {
+		select {
+		case <-time.After(waitTime):
+		case <-d.quitCh:
+		}
+		return
+	}
+
+	// TODO: Do we need to wait for a while?
+	if latestSchemaVersion == 0 {
+		log.Infof("[ddl] schema version doesn't changed")
+		return
+	}
+	// TODO: Make ctx exits when the d is close.
+	ctx, cancelFunc := goctx.WithTimeout(goctx.Background(), waitTime)
+	defer cancelFunc()
+	err := d.worker.updateGlobalVersion(ctx, latestSchemaVersion)
+	if err != nil {
+		log.Infof("[ddl] udpate latest schema version %d failed %v", latestSchemaVersion, err)
+	}
+
+	err = d.worker.checkAllVersions(ctx, latestSchemaVersion)
```

Comment:
If we meet error here, should we return nothing to the caller? What will the caller do?